src/Sticky/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const POSITION_VARIANTS = [
+  'top',
+  'bottom',
+];
+
+const Sticky = ({
+  position,
+  children,
+  offset,
+  className,
+  style,
+}) => {
+  const [isSticky, setIsSticky] = useState(false);
+  const ref = useRef();
+
+  useEffect(() => {
```

Comment:
I believe we may want to be using `useLayoutEffect` here instead of a normal `useEffect`. See [documentation](https://reactjs.org/docs/hooks-reference.html#uselayouteffect) and [this summary](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect#uselayouteffect) for more details.

<img width="676" alt="image" src="https://user-images.githubusercontent.com/2828721/156070495-eaea8a3c-ed6f-4fa6-b458-caa0edc503a1.png">

Because we're taking measurements of offset for a DOM element, I believe this may be one of those rare times to utilize `useLayoutEffect`. What do you think? The main difference with `useLayoutEffect` is that the DOM measurements would be taken before the updates are painted to the screen, which might mean a more smooth shadow while scrolling, for example.

Reply:
Yes, it makes sense, I fix it